File: playground/BOJ/S_boj_1316/boj_1316.py
'''---------------------------------------------
Sub  : [BOJ] 그룹 단어 체커
Link : https://www.acmicpc.net/problem/1316
Level: 실버 5
Tag  : Python, 문자열
Memo
 - 단어의 중복 문제는 set()을 사용하여 풀기
---------------------------------------------'''

import sys
input = sys.stdin.readline

N = int(input().strip())
count = 0

for _ in range(N):
    chars = input().strip()
    seen = set()
    prev_char = None
    is_group_word = True

    for char in chars:
        if char != prev_char:  # 문자가 바뀔 때
            if char in seen:   # 이미 등장했던 문자면 그룹 단어가 아님
                is_group_word = False
                break
            seen.add(char)
        prev_char = char

    if is_group_word:
        count += 1

print(count)

File: Online_Judge_Problems/boj/boj_1316_그룹_단어_체커/boj_1316.c
/*-----------------------------------------------------
Sub  : [BOJ] 그룹 단어 체커
Link : https://www.acmicpc.net/problem/1316
Level: 실버 5
Tag  : C, String
Memo
-----------------------------------------------------*/

#include <stdio.h>
#include <string.h>

// 그룹 단어인지 확인하는 함수
int is_group_word(const char *word) {
    int seen[26] = {0};  // 알파벳 방문 여부 저장
    char prev = 0;

    for (int i = 0; word[i] != '\0'; i++) {
        char current = word[i];
        if (current != prev) {
            if (seen[current - 'a']) {
                return 0;
            }
            seen[current - 'a'] = 1;
        }
        prev = current;
    }
    return 1;
}

int main() {
    int N, count = 0;
    char word[101];

    scanf("%d", &N);
    for (int i = 0; i < N; i++) {
        scanf("%s", word);
        if (is_group_word(word)) {
            count++;
        }
    }

    printf("%d\n", count);
    return 0;
}
